Sources/ComposableArchitecture/Reducer.swift

```diff
@@ -189,6 +189,59 @@ public struct Reducer<State, Action, Environment> {
       return self.reducer(&state!, action, environment)
     }
   }
+    
+    /// Transforms a reducer that works on non-optional state into one that works on optional state by
+    /// only running the non-optional reducer when state is non-nil.
+    ///
+    /// Often used in tandem with `pullback` to transform a reducer on a non-optional local domain
+    /// into a reducer on a global domain that contains an optional local domain:
+    ///
+    ///     // Global domain that holds an optional local domain:
+    ///     struct AppState { var modal: ModalState? }
+    ///     struct AppAction { case modal(ModalAction) }
+    ///     struct AppEnvironment { var mainQueue: AnySchedulerOf<DispatchQueue> }
+    ///
+    ///     // A reducer that works on the non-optional local domain:
+    ///     let modalReducer = Reducer<ModalState, ModalAction, ModalEnvironment { ... }
+    ///
+    ///     // Pullback the local modal reducer so that it works on all of the app domain:
+    ///     let appReducer: Reducer<AppState, AppAction, AppEnvironment> =
+    ///       modalReducer.optional().pullback(
+    ///         state: \.modal,
+    ///         action: /AppAction.modal,
+    ///         environment: { ModalEnvironment(mainQueue: $0.mainQueue) }
+    ///       )
+    ///
+    /// - See also: `IfLetStore`, a SwiftUI helper for transforming a store on optional state into a
+    ///   store on non-optional state.
+    /// - See also: `Store.ifLet`, a UIKit helper for doing imperative work with a store on optional
+    ///   state.
+    public func optional(_ file: String = #file, _ line: Int = #line) -> Reducer<State?, Action, Environment> {
```

Comment:
I think `assertionFailure` parameters are `StaticString` and `Uint`...

```suggestion
    public func optional(_ file: StaticString = #file, _ line: UInt = #line) -> Reducer<State?, Action, Environment> {
```

Reply:
Yes you're totally right
![Screenshot 2020-07-08 at 19 22 56](https://user-images.githubusercontent.com/750912/86950421-74d6cc80-c150-11ea-8ddb-7d470a23e006.png)